activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -614,6 +625,20 @@ def has_default_function?(default_value, default)
         def load_additional_types(oids = nil)
           initializer = OID::TypeMapInitializer.new(type_map)
 
+          if additional_type_records_cache.present?
+            use_cache = if oids.nil?
+              true
+            else
+              cached_oids = additional_type_records_cache.map { |oid| oid["oid"] }
+              (oids - cached_oids).empty?
+            end
+
+            if use_cache
```

Comment:
I think there's a slight bug here if `load_additional_types` is called with a non-nil argument first.  It will return the limited result instead of doing a lookup.  Maybe it's safe to assume it will always be called w/ nil/empty args first though?

Reply:
The first call of `load_additional_types` is inside the `initialize` method without any argument, so I think it's safe to ignore your mentioned case 😉 .